### PR TITLE
feat(api): create new endpoint get vulnerabilities of a component

### PR DESCRIPTION
**Pull request by tuannn2**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

- Run this endpoint to get list vulnerability of a component
- Modify response endpoint when get list vulnerability of a release 

Issue: #1908

### How To Test?
- Run command line: 
```
$ curl 'https://sw360.org/api/components/17653524/vulnerabilities' -i -X GET \
    -H 'Content-Type: application/hal+json' \
    -H 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsic3czNjAtUkVTVC1BUEkiXSwidXNlcl9uYW1lIjoiYWRtaW5Ac3czNjAub3JnIiwic2NvcGUiOlsiYWxsIl0sImV4cCI6MTY4MjUwMzExOCwiYXV0aG9yaXRpZXMiOlsiUkVBRCIsIldSSVRFIl0sImp0aSI6ImsxTGVOZmlodlpnNF9iZjBqb1Q0RkgwT0FRayIsImNsaWVudF9pZCI6InRydXN0ZWQtc3czNjAtY2xpZW50In0.lKSfXikX3g_DzyESF3rOJ1mnM0khnopUesrxXXccxKngx1wKdxN6QP0aznFgW17AgFlcRmlcOfHG34QZTCaP1QwM2Pli6dwBBj9cvSKcOpPEQPt9P8XHOVef-2dgYTbCCKvdzKJmyshgwfyMDP9WwYls5uPVOoEbhitSZylIbOgLLjMqTUbIpbO0cmwBe6DM-q5ifZSFmqMSEk0dhc34mC5VrcXDucf-2tLCp9jhhEia3LdgcMicAa7E6JlJoi9-D6Q8jOYvOwSQ9gaZi726kkol1HAx9fqAon7ubf0XkpBCZOT_udHjKDUVnCEWfwGs_J_BTbE4LZj4QhGsXvYNtA' \
    -H 'Accept: application/hal+json'
```
- Expected Output: 
```

{
  "_embedded" : {
    "sw360:vulnerabilityDTOes" : [ {
      "externalId" : "12345",
      "title" : "12345_Title",
      "priority" : "2 - major",
      "releaseVulnerabilityRelation" : {
        "releaseId" : "3765276512",
        "vulnerabilityId" : "1333333333",
        "verificationStateInfo" : [ {
          "checkedOn" : "2018-12-18",
          "checkedBy" : "admin@sw360.org",
          "comment" : "Change status not checked",
          "verificationState" : "NOT_CHECKED"
        } ],
        "matchedBy" : "matchedBy",
        "usedNeedle" : "usedNeedle"
      },
      "projectRelevance" : "IRRELEVANT",
      "comment" : "Lorem Ipsum",
      "projectAction" : "Update to Fixed Version",
      "intReleaseId" : "3765276512",
      "intReleaseName" : "Angular 2.3.0"
    }, {
      "externalId" : "23105",
      "title" : "23105_Title",
      "priority" : "1 - critical",
      "releaseVulnerabilityRelation" : {
        "releaseId" : "3765276512",
        "vulnerabilityId" : "122222222",
        "verificationStateInfo" : [ {
          "checkedOn" : "2016-12-18",
          "checkedBy" : "user@sw360.org",
          "comment" : "Change status checked",
          "verificationState" : "CHECKED"
        } ],
        "matchedBy" : "matchedBy1",
        "usedNeedle" : "usedNeedle1"
      },
      "projectRelevance" : "APPLICABLE",
      "comment" : "Lorem Ipsum",
      "projectAction" : "Update to Fixed Version",
      "intReleaseId" : "3765276512",
      "intReleaseName" : "Angular 2.3.0"
    } ]
  },
  "_links" : {
    "curies" : [ {
      "href" : "https://sw360.org/docs/{rel}.html",
      "name" : "sw360",
      "templated" : true
    } ]
  }
}
```

### CODE REVIEW

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java [hunk 1]
```diff
@@ -632,20 +632,14 @@ static abstract class ObligationMixin extends Obligation {
                 "id",
                 "revision",
                 "type",
-                "title",
```
Comment (smrutis1) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java:635]: These fields have been removed from blacklist. But I cannot see those in the response example you have shared. Is there anything I am missing here?

**Review by smrutis1 — COMMENTED** [review]
One query

**Comment by tuannn2** [comment]
Thank you for your reviewing, @smrutis1
I have updated the response title.
Please check it !

**Review by smrutis1 — APPROVED** [review]
Code looks good

**Pull request merged**